tests/test.conf

```diff
@@ -1497,21 +1497,21 @@
 -v 3
 -l PSK-AES256-GCM-SHA384
 
-# server TLSv1.2 AES128-CCM-8
+# server TLSv1.2 AES128-CCM8
```

Comment:
Where are we testing the "old" name?

Also, can the alias code be removed easily by those who don't want it?

Reply:
John and I independently tested the old name variants, with manipulated codebases.

Currently unit.test silently ignores not-found suites, so it's not a viable test for the alias mechanism.  John and I discussed this shortcoming today, and we want to fix it, but it's nontrivial to refactor the unit.test framework to do so.

At an intermediate stage of development I had included several of the "CCM-8" variants in tests/test.conf, annotated as "(compat alias test)" or somesuch.  Not a very meaningful test due to unit.test's shortcomings, but better than nothing.


If someone wanted to remove the aliases, they'd simply remove the `SUITE_ALIAS()` lines in the `cipher_names` definition.  Unfortunately the comma is outside the parens, so they'd need to remove the whole line, not just do `#define SUITE_ALIAS(...)`, but it's obvious what to do.

Reply:
What's the point of alias code that doesn't work (I don't see a test that shows this and can be maintained), can't be tested, and can't be removed with a simple define?

Reply:
Are you sure CCM-8 can't be tested here w/o code changes?

Reply:
Those are all good points.  I'll add convenient gating on the alias code, and I'll find the right spot(s) in the code to affirmatively exercise the alias functionality.  By definition this will require a test that reports unexpected suite-not-found errors, but surely this class of test already exists somewhere in the codebase.

Reply:
Not exactly.  If both the old and new names are supported why can't we simply test both names.  Not found cipher suites are not expected and do not need to be added to tests.